core/src/main/java/bio/terra/pearl/core/model/address/AddressValidationResultDto.java

```diff
@@ -1,41 +1,36 @@
 package bio.terra.pearl.core.model.address;
 
-import bio.terra.pearl.core.model.participant.MailingAddress;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 import lombok.experimental.SuperBuilder;
 
 import java.util.List;
-import java.util.UUID;
 
 @Getter
 @Setter
 @SuperBuilder
 @NoArgsConstructor
 public class AddressValidationResultDto {
-    // If false, property indicates strong failure -
-    // the address was incomplete or invalid and an
-    // improved address could not be found.
+    // If true, indicates that there is a valid address which
+    // could be found.
+    // (makes no claim about the validity of the provided address)
     private boolean valid;
 
-    // This property provides a suggestion for
-    // an improved mailing address; typically,
-    // this means the address was valid, but
-    // could be improved
+    // This property indicates a valid mailing address provided
+    // by USPS/international databases
     private MailingAddress suggestedAddress;
 
     // Components which were missing from
     // an invalid address
-    // see: https://developers.google.com/maps/documentation/places/web-service/supported_types#table2
-    //      for possible values
-    private List<String> missingComponents;
+    private List<AddressComponent> invalidComponents;
 
-    // Raw string values of things that were not able to
-    // resolve, e.g. "12345678" in "12345678 Main St"
-    private List<String> unresolvedTokens;
+    // Indicates if the suggested address required
+    // inferring components of the address that
+    // were either missing or incorrect
+    private Boolean hasInferredComponents;
 
-    // When validating the same address over and over again,
-    // use the same sessionId
-    private UUID sessionId;
```

Comment:
@devonbush fyi after playing around with the Java client for Smarty I decided to go ahead and change our result dto to match as it wouldn't be too much work. Should be similar-ish behavior to what you reviewed before. The only real new concept is `hasInferredComponents`. I'm not using the `vacant` field as of yet, but it could be used as a warning in the future (we could expand our concept of warnings, as Smarty has a lot of not-quite-valid but not-quite-invalid states it can return, vacant just being one of them).

Also, after talking with Erin, we're thinking about having the street1 input be a search which autocompletes the rest of the address. The service we're using supports that so I created a ticket to stub that and add it to the admin ui. After all that prep work (whew) it _should_ be pretty easy to connect up the UI to the real service with the testing code I was playing around with this morning.

Reply:
nice, 100% agree